kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PaymentRequestDocument.java

```diff
@@ -0,0 +1,189 @@
+            
+            if (!docAuth.isAuthorizedByTemplate(this, 
+            		KFSConstants.CoreModuleNamespaces.KFS, 
+            		KFSConstants.PermissionTemplate.EDIT_BANK_CODE.name,
+            		GlobalVariables.getUserSession().getPrincipalId())) {
+                synchronizeBankCodeWithPaymentMethod();
+            } else {
+                // ensure that the name is updated properly
+                refreshReferenceObject(BANK);
+            }
+        }
+    }
+
+    public void synchronizeBankCodeWithPaymentMethod() {
+        Bank bank = getPaymentMethodGeneralLedgerPendingEntryService().getBankForPaymentMethod(getPaymentMethodCode());
+        if (bank != null) {
+            setBankCode(bank.getBankCode());
+            setBank(bank);
+        }
+        else {
+            // no bank code, no bank needed
+            setBankCode(null);
+            setBank(null);
+        }
+    }
+
+    protected PaymentMethodGeneralLedgerPendingEntryService getPaymentMethodGeneralLedgerPendingEntryService() {
+        if (paymentMethodGeneralLedgerPendingEntryService == null) {
+            paymentMethodGeneralLedgerPendingEntryService = SpringContext.getBean(PaymentMethodGeneralLedgerPendingEntryService.class);
+        }
+        return paymentMethodGeneralLedgerPendingEntryService;
+    }
+    
+    @Override
+    public void customizeExplicitGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySourceDetail postable, GeneralLedgerPendingEntry explicitEntry) {
+        super.customizeExplicitGeneralLedgerPendingEntry(postable, explicitEntry);
+        // if the document is not processed using PDP, then the cash entries
+        // need to be created instead of liability
+        // so, switch the document type so the offset generation uses a cash
+        // offset object code
+        if (!getPaymentMethodGeneralLedgerPendingEntryService().isPaymentMethodProcessedUsingPdp(getPaymentMethodCode())) {
+            explicitEntry.setFinancialDocumentTypeCode(DOCUMENT_TYPE_NON_CHECK);
+        }
+    }
+
+    /**
+     * Update to baseline method to additionally set the payment method when a vendor is selected.
+     */
+    @Override
+    public void populatePaymentRequestFromPurchaseOrder(PurchaseOrderDocument po, HashMap<String, ExpiredOrClosedAccountEntry> expiredOrClosedAccountList) {
+        super.populatePaymentRequestFromPurchaseOrder(po, expiredOrClosedAccountList);
+        if (ObjectUtils.isNotNull(po.getVendorDetail()) && ObjectUtils.isNotNull(po.getVendorDetail().getExtension())) {
+            if (po.getVendorDetail().getExtension() instanceof VendorDetailExtension && StringUtils.isNotBlank(((VendorDetailExtension) po.getVendorDetail().getExtension()).getDefaultB2BPaymentMethodCode())) {
+                setPaymentMethodCode(((VendorDetailExtension) po.getVendorDetail().getExtension()).getDefaultB2BPaymentMethodCode());
+                synchronizeBankCodeWithPaymentMethod();
+            }
+        }
+    }
+
+    @Override
```

Comment:
These two methods are code for KFSI-3299, but not all of it. This is another bug that should not have been included as part of this mod.